tracardi/process_engine/action/v1/converters/base64/encode/plugin.py

```diff
@@ -0,0 +1,78 @@
+from base64 import b64encode
+
+from tracardi.service.plugin.domain.register import Documentation, Form, FormComponent, FormField, FormGroup, \
+    MetaData, Plugin, PortDoc, Spec
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.runner import ActionRunner
+from .model.models import Configuration
+
+
+def validate(config: dict):
+    return Configuration(**config)
+
+
+class Base64EncodeAction(ActionRunner):
+    config: Configuration
+
+    async def set_up(self, init):
+        self.config = validate(init)
+
+    async def run(self, payload: dict, in_edge=None):
+        dot = self._get_dot_accessor(payload)
+        text = dot[self.config.to_base64]
+        result = b64encode(text.encode(self.config.input_encoding)).decode('ascii')
+
+        return Result(port="payload", value={"base64": result})
+
+
+def register() -> Plugin:
+    return Plugin(
+        start=False,
+        spec=Spec(
+            module=__name__,
+            className=Base64EncodeAction.__name__,
+            inputs=['payload'],
+            outputs=['payload'],
+            version='0.1',
+            license='MIT',
+            author='knittl',
+            manual='base64_encode',
+            init={
+                'to_base64': 'payload@data',
+                'input_encoding': 'utf-8',
+            },
+            form=Form(groups=[
+                FormGroup(
+                    fields=[
+                        FormField(
+                            id='to_base64',
+                            name='String to encode as base64',
+                            description='Type path to string or string itself. Default source is payload.',
+                            component=FormComponent(type='dotPath', props={'defaultSourceValue': 'payload'}),
+                        ),
+                        FormField(
+                            id='input_encoding',
+                            name='Encoding',
+                            description='Input text encoding. Will be used to convert input text to bytes before '
+                                        'base64-encoding it. Set to "utf-8" if unsure.',
+                            component=FormComponent(type='text', props={'label': 'encoding'}),
```

Comment:
Please see the review from the other PR. encodings could be a select component.

Reply:
See my reasoning for why no select component: list would be too long and no freeform input is allowed. I'm open for suggestions, e.g. provide select with the top 10 most common encodings for any special encodings users must go to raw JSON config.

Reply:
Your solution is fine. The most popular are ok.

Reply:
"most popular" is not so easy to answer. I picked the following encodings for now:

* utf_8
* utf_16_be
* utf_16_le
* utf_32_be
* utf_32_le
* ascii
* big5
* euc_jp
* euc_kr
* gbk
* hz
* iso8859-15
* windows-1250

Feel free to suggest other encodings to add or if I should remove some.